src/main/java/org/generationcp/middleware/dao/ims/LotDAO.java

```diff
@@ -450,20 +457,54 @@ public long countSearchLots(final LotsSearchDto lotsSearchDto) {
 		}
 	}
 
-	public List<UserDefinedField> getGermplasmAttributeTypes(final LotsSearchDto searchDto) {
+	public List<Variable> getGermplasmAttributeVariables(final LotsSearchDto searchDto, final String programUUID) {
 		try {
 			final StringBuilder lotsQuery = new StringBuilder(SearchLotDaoQuery.getSelectBaseQuery());
 			SearchLotDaoQuery.addSearchLotsQueryFiltersAndGroupBy(new SqlQueryParamBuilder(lotsQuery), searchDto);
 
-			final String sql = "select distinct {u.*} from atributs a inner join udflds u "
-				+ " 	inner join (" + lotsQuery + ") lots on lots.gid = a.gid"
-				+ " where a.atype = u.fldno"
-				+ " order by u.fname";
-
-			final SQLQuery query = this.getSession().createSQLQuery(sql);
-			SearchLotDaoQuery.addSearchLotsQueryFiltersAndGroupBy(new SqlQueryParamBuilder(query), searchDto);
-			query.addEntity("u", UserDefinedField.class);
-			return query.list();
+			final SQLQuery sqlQuery = this.getSession().createSQLQuery("select "
```

Comment:
@cuenyad sounds the query is very similar to the one I have asked you to move to GermplasmDao
If I have to solve the problem, I will get the gids that results from LotSearchDto and reuse the function in GermplasmDao so we avoid maintaining these 2 queries